dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -33,8 +37,27 @@ def all
     alias_method provider.to_sym, :all
   end
 
+  OAUTH_PARAMS_TO_STRIP = %w{oauth_token oauth_refresh_token}
+
+  def self.get_cache_key(oauth_param, user)
+    "#{oauth_param}_#{user.email}"
+  end
+
   private
 
+  def move_oauth_params_to_cache(user)
+    # Because some oauth tokens are quite large, we strip them from the session
+    # variables and pass them through via the cache instead - they are pulled out again
+    # from User::new_with_session
+    OAUTH_PARAMS_TO_STRIP.each do |param|
+      param_value = user.attributes['properties'][param]
+      user.attributes['properties'][param] = nil
+      cache = CDO.shared_cache
```

Comment:
This (initializing `cache`) could be pulled outside the loop? Assuming so, there could be an early exit if `cache` is `nil`, removing the need to conditionally write to the cache.